### ISSUE TITLE
Windows gettimeofday() should not be exported from haicrypt

### ISSUE THREAD

**Issue by ethouris** [Haivision/srt#59]
I know this works, but this is logically wrong.

The logical split of this is that the common stuff, including "compat" or "platform dependent" things is a dependency of both haicrypt and srt, no matter that haicrypt is also a dependency of srt. So, for example, when srt is using gettimeofday() - and its crafted version on Windows - it uses part of the "platform compat" stuff, which has nothing to do with haicrypt.

There are two possible approaches, which don't break the logic of this whole breakdown:

* Keep all libraries compiled to separate files. However this means that the "compat" stuff must be compiled into yet another library, say "srtcompat". This will be then dependency of haicrypt and srt, independently, and it will result in having 3 library files.
* Keep everything in one library file. Actually haicrypt is a kinda private dependency of srt and there's no need for it to be exposed (actually even its public headers are not installable). In this case, all 3 things can be maintained as separate subprojects, but everything can be compiled into one library.

Consider both solutions. Remember though that the latter is simple, simplifies installation and deployment, and, what is most important, gettimeofday() or any other compat stuff doesn't need to be exported at all.

**Comment by alexpokotilo** [comment]
Hi,
nobody will use haicrypt without SRT so merge is the best and simplest solution.

**Comment by russelltg** [comment]
That should fix the build for #58 as well

**Comment by ethouris** [comment]
I've prepared a solution for that and I'm about to give it up to merge. The CMakeLists.txt file is now back one central (the definitions concerning haicrypt and srtcore are now intermixed due to CMake limitations, and can't be kept in two separate files).

Russel: I may need your further help after this thing. I haven't touched anything that concerns the things mentioned in the topic - that is, exporting gettimeofday(). The macro for declspec for the default mode probably will have to be set "more properly". At least I have compiled this on Windows MSVS 2013.
